### PR TITLE
CI: check for broken links on website

### PR DESCRIPTION
**Pull request by strager**
Ensure all internal links on our website are working.
    
External links are not checked to avoid flakiness.

### CODE REVIEW

**Comment by strager** [comment]
@emadflash Can you take a look at this PR and see if anything looks wrong?

File: .github/workflows/build-website.yml [hunk 1]
```diff
@@ -40,6 +40,29 @@ jobs:
...
         run: cd website && zip -r ../website.zip www
 
+      - name: set up check-links
+        run: |
+          python3.7 -m venv ENV
+          ENV/bin/pip install bs4 requests
+      - name: check links (dev)
+        run: |
+          cd website/
+          yarn start localhost 9001 &
```
Comment (emadflash) [inline .github/workflows/build-website.yml:50]: why start the server inside website/ ? it should be website/public
something is wrong, its not reporting errors that way

**Review by emadflash — CHANGES_REQUESTED** [review]

File: .github/workflows/build-website.yml (hunk 1 above)
Comment (strager) [reply .github/workflows/build-website.yml:50]: When I develop the website, I always run `yarn start` from within website/.

The website/public/ directory contains the page sources, but doesn't contain the server entry point. The server entry point is website/run-server.mjs.

Comment (emadflash) [reply .github/workflows/build-website.yml:50]: ok got. so yarn sets the entry point. i was using `python3 -m http.server 8080` inside /website/

Comment (strager) [reply .github/workflows/build-website.yml:50]: Ah, I see why you were confused. The website's guts have been restructured since you made check-links.

**Pull request merged**